modules/imgproc/src/sumpixels.cpp

```diff
@@ -147,7 +147,9 @@ struct Integral_SIMD<uchar, int, double>
                 v_expand(el8, el4l, el4h);
                 el4l += prev;
                 el4h += el4l;
-                prev = vx_setall_s32(v_rotate_right<v_int32::nlanes - 1>(el4h).get0());
+
+                prev = vx_setall_s32(v_extract_n<v_int32::nlanes - 1>(el4h));
```

Comment:
Probably more specific intrinsic fits better in this case:
- something like `v_broadcast_element<v_int32::nlanes - 1>(el4h)`
- can be implemented through permutation/shuffle

Reply:
Yes.

Plain C++ code:
```
/** Broadcast i-th element of vector
Scheme:
@code
{ v[0] v[1] v[2] ... v[SZ] } => { v[i], v[i], v[i] ... v[i] }
@endcode
Restriction: 0 <= i < nlanes (supported: TBD)
Supported types: TBD (lets start from `int32_t` - make sense after revieving of real intrinsic support)
 */
template<int i, typename _Tp, int n>
inline v_reg<_Tp, n> v_broadcast_element(const v_reg<_Tp, n>& a)
{
    return v_reg<_Tp, n>::all(a.s[i]);
}
```

Do you see any objections?